protocol/src/protocol.callHierarchy.proposed.ts

```diff
@@ -0,0 +1,172 @@
+         * Capabilities specific to the `textDocument/callHierarchy`
+         */
+        callHierarchy?: {
+            /**
+             * Whether implementation supports dynamic registration. If this is set to `true`
+             * the client supports the new `(TextDocumentRegistrationOptions & StaticRegistrationOptions)`
+             * return value for the corresponding server capability as well.
+             */
+            dynamicRegistration?: boolean;
+        };
+    }
+}
+
+export interface CallHierarchyServerCapabilities {
+    /**
+     * The server provides Call Hierarchy support.
+     */
+    callHierarchyProvider?: boolean | (TextDocumentRegistrationOptions & StaticRegistrationOptions);
+}
+
+/**
+ * Request to request the call hierarchy at a given text document position.
+ *
+ * The request's parameter is of type [CallHierarchyParams](#CallHierarchyParams). The response
+ * is of type [CallHierarchyItem](#CallHierarchyItem) or a Thenable that resolves to such.
+ *
+ * The optional request's parameter defines the maximum number of levels to [resolve](#CallHierarchyParams.resolve) by this request.
+ * Unresolved items can be resolved in subsequent `callHierarchy/resolve` requests.
+ */
+export namespace CallHierarchyRequest {
+    export const type = new RequestType<CallHierarchyParams, CallHierarchyItem, void, TextDocumentRegistrationOptions>('textDocument/callHierarchy');
+    export type HandlerSignature = RequestHandler<CallHierarchyParams, CallHierarchyItem | null, void>;
+}
+
+/**
+ * Request to resolve a call hierarchy item.
+ *
+ * The request's parameter is of type [ResolveCallHierarchyItemParams](#ResolveCallHierarchyItemParams). The response
+ * is of type [CallHierarchyItem](#CallHierarchyItem) or a Thenable that resolves to such.
+ */
+export namespace CallHierarchyResolveRequest {
+    export const type = new RequestType<ResolveCallHierarchyItemParams, CallHierarchyItem, void, void>('callHierarchy/resolve');
+    export type HandlerSignature = RequestHandler<ResolveCallHierarchyItemParams, CallHierarchyItem | null, void>;
+}
+
+/**
+ * The parameters of a `textDocument/callHierarchy` request.
+ */
+export interface CallHierarchyParams extends TextDocumentPositionParams {
+    /**
+     * The number of levels to resolve.
+     */
+    resolve?: number;
+    /**
+     * The direction of calls to resolve.
+     */
+    direction?: CallHierarchyDirection;
+}
+
+/**
+ * The parameters of a `callHierarchy/resolve` request.
+ */
+export interface ResolveCallHierarchyItemParams {
+    /**
+     * Unresolved item.
+     */
+    item: CallHierarchyItem;
+    /**
+     * The number of levels to resolve.
+     */
+    resolve: number;
+    /**
+     * The direction of calls to resolve.
+     */
+    direction: CallHierarchyDirection;
+}
+
+/**
+ * The direction of a call hierarchy.
+ */
+export namespace CallHierarchyDirection {
+    /**
+     * The callers of a symbol.
+     */
+    export const Incoming: 1 = 1;
```

Comment:
Can you please explain it to me why does it start from `1` instead of `0`? Most of the values start from `1`, but for instance, `TextDocumentSyncKind` starts with `0`.

The only pattern I have seen so far is when a value is optional on the use-site, then it can start from `0`. Otherwise, it starts with a non-zero integer, `1`. Thanks!

See: https://github.com/Microsoft/vscode-languageserver-node/pull/426/files#diff-1b6afc80d5099ec4d774052b71cc1d10R87

Reply:
> Can you please explain it to me ...

I can't. I looked into the `protocol.ts` and found a couple of examples where it starts from `1`, and aligned accordingly.

Reply:
@dbaeumer, could you please advise? Thank you!

Reply:
New enums should start with 1. We have some that start with 0 (old once) and I don't want to break.